decidim-core/lib/tasks/upgrade/decidim_fix_nickname_uniqueness.rake

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :upgrade do
+    desc "Modify nicknames with random numbers when exists similar ones case insensitively"
+    task fix_nickname_uniqueness: :environment do
+      logger = Logger.new($stdout)
+      logger.info("Updating users nickname ...")
+
+      # list of users already changed in the process
+      has_changed = []
+
+      Decidim::User.find_each do |user|
+        next if has_changed.include? user
+
+        Decidim::User.where("nickname ILIKE ?", user.nickname.downcase).order(:created_at).each do |similar_user|
```

Comment:
Please also add the organization criteria here, i.e. matching only in the same organization as `user`.